### PR TITLE
Add new `Lint/UselessNumericOperation` cop

### PR DESCRIPTION
**Pull request by Zopolis4**
> [!NOTE]
> The operations with constant results (i.e. `x * 0` will be implemented in a different cop, as they include patterns like `x - x` which will require modifications to `Lint/BinaryOperatorWithIdenticalOperands`)

Makes progress on #11191, taking care of the inconsequential operations.

Certain numeric operations have no impact, being:
Adding or subtracting 0, multiplying or dividing by 1 or raising to the power of 1.

These are probably mistakes or leftover from debugging, so flag and optionally remove them.

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/lint/useless_numeric_operation.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      #
+      #   # bad
+      #   x += 0
+      #   x -= 0
+      #   x *= 1
+      #   x /= 1
+      #   x **= 1
+      #
+      #   # good
+      #   x = x
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/useless_numeric_operation.rb:30]: Perhaps you meant to write just `x` here, as I don't see how `x = x` makes sense in this context.

Comment (Zopolis4) [reply lib/rubocop/cop/lint/useless_numeric_operation.rb:30]: My thinking was to only remove the numeric component-- the assignment can be handled by `Lint/UselessAssignment`.

**Comment by bbatsov** [comment]
What about doing the same things with floats? The results will be similar, even if often they won't be totally accurate due to the way floats are implemented internally.

**Comment by Zopolis4** [comment]
I guess? To be honest, I'm not confident enough in my understanding of floats to implement that.

**Comment by bbatsov** [comment]
Okay, I guess we can merge this as is. Thanks for working on it!

**Pull request merged**